blockservice/blockservice.go

```diff
@@ -31,6 +32,7 @@ type BlockService interface {
 	GetBlock(ctx context.Context, c *cid.Cid) (blocks.Block, error)
 	GetBlocks(ctx context.Context, ks []*cid.Cid) <-chan blocks.Block
 	DeleteBlock(o blocks.Block) error
+	NewSession(context.Context) *Session
```

Comment:
Couldn't this just be a helper function?

```go
func NewSession(ctx context.Context, bs BlockService) *Session {
	exchange := bs.Exchange()
	if bswap, ok := exchange.(*bitswap.Bitswap); ok {
		ses := bswap.NewSession(ctx)
		return &Session{
			ses: ses,
			bs:  bs.Blockstore(),
		}
	}
	return &Session{
		ses: bs.exchange,
		bs:  bs.Blockstore(),
	}
}
```

(we could also make this non-bitswap specific by adding a `NewSession(context.Context) exchange.Interface` method to `exchange.Interface` but we don't need that now).